pkg/kds/v2/client/kds_client.go

```diff
@@ -23,12 +25,14 @@ func (u *UpstreamResponse) Validate() error {
 	if u.AddedResources == nil {
 		return nil
 	}
+	var err error
 	for _, res := range u.AddedResources.GetItems() {
-		if err := model.Validate(res); err != nil {
-			return err
```

Comment:
We can just add 
```text
u.InvalidResourcesKey = append(u.InvalidResourcesKey, core_model.MetaToResourceKey(res.GetMeta()))
```
on top of `return err`

There's no need to introduce `errors.Join` here right?

Reply:
It's good for NACK message so the user knows that there is an issue with the resource